src/Core/AdminConsole/Services/Implementations/PolicyService.cs

```diff
@@ -124,10 +124,16 @@ public class PolicyService : IPolicyService
                 switch (policy.Type)
                 {
                     case PolicyType.TwoFactorAuthentication:
-                        foreach (var orgUser in removableOrgUsers)
+                        foreach (var orgUser in removableOrgUsers.OrderBy(ou => !ou.HasMasterPassword))
```

Comment:
This appears to be sorting the list in the reverse order that you expect. You can remove the `!` which will sort it in ascending order by default (false values will come first). I would also recommend adding a comment to explain why the order is important.

Reply:
You're absolutely right, the unit tests confirmed that!